sympy/stats/crv.py

```diff
@@ -693,7 +693,8 @@ def compute_density(self, expr, **kwargs):
             raise ValueError("Can not solve %s for %s"%(expr, self.value))
         fx = self.compute_density(self.value)
         fy = sum(fx(g) * abs(g.diff(y)) for g in gs)
-        return Lambda(y, fy)
+        c_domain = continuous_domain(fy, y, S.Reals)
+        return Lambda(y, Piecewise((fy, c_domain.contains(y)), (0, True)))
```

Comment:
ping @Upabjojr Is this good? Though it will slow down things a little bit but the results will be correct provided `continuous_domain` generates the correct domain.

Reply:
Slowdowns are OK with me. I prefer to have correct results.

Isn't it a repetition of the `Piecewise` in `Distribution.__call__` ?

Reply:
> Isn't it a repetition of the `Piecewise` in `Distribution.__call__` ?

For an RV input `density` gives an instance of `Distribution` and for an expression containing RV,  a `Lambda` object is returned, which is inconsistent I believe. I think, instead of putting domain check here, we can return a abstract `Distribution` object which will have a `__call__` method and domain check can be added there.

Reply:
```python
In [1]: from sympy.stats import *

In [2]: N = Normal('N', 0, 1)

In [3]: density(N)
Out[3]: NormalDistribution(0, 1)

In [4]: density(exp(N))
Out[4]: 
            2    
        -log (y) 
        ─────────
            2    
    √2⋅ℯ         
y ↦ ─────────────
       2⋅√π⋅│y│
```